src/stan/error_handling/matrix/check_spsd_matrix.hpp

```diff
@@ -15,21 +15,25 @@ namespace stan {
      * Return <code>true</code> if the specified matrix is a 
      * square, symmetric, and positive semi-definite.
      *
-     * @param function
-     * @param name
-     * @param y Matrix to test.
+     * @tparam T Scalar type of the matrix
+     *
+     * @param function Function name (for error messages)
+     * @param name Variable name (for error messages)
+     * @param y Matrix to test
+     *
      * @return <code>true</code> if the matrix is a square, symmetric,
-     * and positive semi-definite.
-     * @return throws if any element in matrix is nan.
-     * @tparam T Type of scalar.
+     *   and positive semi-definite.
+     * @throw <code>std::invalid_argument</code> if the matrix is not square
+     *   or if the matrix is 0x0
+     * @throw <code>std::domain_error</code> if the matrix is not symmetric
+     *   or if the matrix is not positive semi-definite
      */
-    // FIXME: update warnings
     template <typename T_y>
     inline bool check_spsd_matrix(const std::string& function, 
                                   const std::string& name,
                                   const Eigen::Matrix<T_y,Eigen::Dynamic,Eigen::Dynamic>& y) {
       check_square(function, name, y);
-      check_positive(function, "rows", y.rows());
+      check_positive_index(function, name, "rows()", y.rows());
```

Comment:
Why is this called `check_positive_index` when what's being check is not an index but a size?